prometheus/src/main/scala/filodb/prometheus/ast/Vectors.scala

```diff
@@ -87,6 +93,11 @@ trait Vectors extends Scalars with TimeUnits with Base {
           case NotEqual(false) => ColumnFilter(labelMatch.label, query.Filter.NotEquals(labelMatch.value))
           case other: Any      => throw new IllegalArgumentException(s"Unknown match operator $other")
         }
+      }.filterNot(_.column == ColumnSelectorLabel)
+
+    protected def labelMatchesToColumnName(labels: Seq[LabelMatch]): Seq[String] =
+      labels.collect {
+        case LabelMatch(ColumnSelectorLabel, EqualMatch, col) => col
```

Comment:
EqualMatch only at the moment. We may support multiple columns with the regex filter or some other custom "in" operator that we can specify in PromQL.